esmvalcore/preprocessor/_area.py

```diff
@@ -84,8 +97,19 @@ def extract_region(cube, start_longitude, end_longitude, start_latitude,
         select_lats = (lats >= start_latitude) | (lats <= end_latitude)
 
     selection = select_lats & select_lons
-    selection = da.broadcast_to(selection, cube.shape)
-    cube.data = da.ma.masked_where(~selection, cube.core_data())
+
+    # Crop the selection, but keep rectangular shape
+    i_range, j_range = selection.nonzero()
+    if i_range.size == 0:
+        raise ValueError("No data points available in selected region")
+    i_min, i_max = i_range.min(), i_range.max()
+    j_min, j_max = j_range.min(), j_range.max()
+    i_slice, j_slice = slice(i_min, i_max + 1), slice(j_min, j_max + 1)
+    cube = cube[..., i_slice, j_slice]
```

Comment:
Maybe use `cube.coord_dims` to determine the correct axes?

Reply:
The CMOR conventions specify that the last two axes are latitude and longitude, right?

Reply:
Yeah, I think that's right. I was unsure with land use types etc. but a quick check seems to confirm it.